### PR TITLE
Support multiple databases for PostgreSQL.

### PR DESCRIPTION
**Pull request by kfwang90**
Fixes about #8736.

### CODE REVIEW

File: shardingsphere-features/shardingsphere-ha/shardingsphere-ha-provider/shardingsphere-ha-mgr/pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <modelVersion>4.0.0</modelVersion>
     <parent>
         <groupId>org.apache.shardingsphere</groupId>
-        <artifactId>shardingsphere-ha</artifactId>
+        <artifactId>shardingsphere-ha-provider</artifactId>
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-ha/shardingsphere-ha-provider/shardingsphere-ha-mgr/pom.xml:25]: What is the proposal here?

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-ha/shardingsphere-ha-provider/shardingsphere-ha-mgr/pom.xml (hunk 1 above)
Comment (kfwang90) [reply shardingsphere-features/shardingsphere-ha/shardingsphere-ha-provider/shardingsphere-ha-mgr/pom.xml:25]: > What is the proposal here?

When I use maven clean intall all projects, it has a error at here. And this project is the child of shardingsphere-ha-provider. So I modify it and install again, thers is no more errors.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/metadata/dialect/PostgreSQLDataSourceMetaData.java [hunk 2]
```diff
@@ -41,7 +41,7 @@
     
     private final String schema;
     
-    private final Pattern pattern = Pattern.compile("jdbc:postgresql://([\\w\\-\\.]+):?([0-9]*)/([\\w\\-]+)", Pattern.CASE_INSENSITIVE);
+    private final Pattern pattern = Pattern.compile("jdbc:postgresql://([\\w\\-\\.]+):?([0-9]*),?.*?/([\\w\\-]+)?\\S*", Pattern.CASE_INSENSITIVE);
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/metadata/dialect/PostgreSQLDataSourceMetaData.java:44]: What is the different for use multiple hosts and replica-query rule?

We need to discuss the production design here. It is better to use same way for same feature. I prefer use replica-query rule to deal with all master-slave scenarios.

Comment (kfwang90) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/metadata/dialect/PostgreSQLDataSourceMetaData.java:44]: Like this secene: There is PG clusters, contains one master node and another async slave node. 
The replica-query file can config as this:
**master : jdbc:postgresql://masterIp:5432,slaveIp:5432/db?targerServerType=master**
**slave : jdbc:postgresql://masterIp:5432,slaveIp:5432/db?targerServerType=preferSlave**
When PG master node goes down, I can promote async node change to new master fast. And the replica-query config file don't have to change. And the application don't have to restart. JDBC will recovery over a moment time.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**